File: LABORATORIO_02/Propuestos/Propuestos_02/src/main/java/ps/lab/Cajero.java
package ps.lab;

import java.util.Scanner;

public class Cajero {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        CajeroAutomatico cajero = new CajeroAutomatico(1000.0);

        int opcion;
        do {
            System.out.println("\n=== MENÚ CAJERO AUTOMÁTICO ===");
            System.out.println("1. Consultar Saldo");
            System.out.println("2. Depositar Dinero");
            System.out.println("3. Retirar Dinero");
            System.out.println("4. Salir");
            System.out.print("Seleccione una opción: ");

            while (!scanner.hasNextInt()) {
                System.out.println("Entrada inválida. Intente de nuevo:");
                scanner.next(); // descarta entrada inválida
            }
            opcion = scanner.nextInt();

            switch (opcion) {
                case 1:
                    System.out.printf("Saldo actual: S/.%.2f%n", cajero.consultarSaldo());
                    break;
                case 2:
                    System.out.print("Ingrese monto a depositar: ");
                    if (scanner.hasNextDouble()) {
                        double deposito = scanner.nextDouble();
                        try {
                            cajero.depositar(deposito);
                            System.out.println("Depósito realizado con éxito.");
                        } catch (IllegalArgumentException e) {
                            System.out.println("Error: " + e.getMessage());
                        }
                    } else {
                        System.out.println("Entrada inválida.");
                        scanner.next(); // descarta entrada inválida
                    }
                    break;
                case 3:
                    System.out.print("Ingrese monto a retirar: ");
                    if (scanner.hasNextDouble()) {
                        double retiro = scanner.nextDouble();
                        try {
                            cajero.retirar(retiro);
                            System.out.println("Retiro realizado con éxito.");
                        } catch (IllegalArgumentException e) {
                            System.out.println("Error: " + e.getMessage());
                        }
                    } else {
                        System.out.println("Entrada inválida.");
                        scanner.next(); // descarta entrada inválida
                    }
                    break;
                case 4:
                    System.out.println("Gracias por usar el cajero automático. ¡Hasta pronto!");
                    break;
                default:
                    System.out.println("Opción inválida. Intente nuevamente.");
            }

        } while (opcion != 4);

        scanner.close();
    }
}

File: LABORATORIO_02/Propuestos/Propuesto_Python/cajero.py
saldo = 1000.0

def mostrar_menu():
    print("\n--- Menú Cajero Automático ---")
    print("1. Consultar Saldo")
    print("2. Depositar Dinero")
    print("3. Retirar Dinero")
    print("4. Salir")

def consultar_saldo():
    print(f"Tu saldo actual es: S/.{saldo:.2f}")

def depositar():
    global saldo
    try:
        monto = float(input("Ingresa el monto a depositar: S/."))
        if monto > 0:
            saldo += monto
            print(f"Has depositado S/.{monto:.2f}. Nuevo saldo: S/.{saldo:.2f}")
        else:
            print("El monto debe ser mayor que cero.")
    except ValueError:
        print("Por favor, ingresa un número válido.")

def retirar():
    global saldo
    try:
        monto = float(input("Ingresa el monto a retirar: S/."))
        if monto <= 0:
            print("El monto debe ser mayor que cero.")
        elif monto > saldo:
            print("Fondos insuficientes.")
        else:
            saldo -= monto
            print(f"Has retirado S/.{monto:.2f}. Nuevo saldo: S/.{saldo:.2f}")
    except ValueError:
        print("Por favor, ingresa un número válido.")

def main():
    print("Bienvenido al Cajero Automático")
    print("Tu saldo inicial es: S/.1000.00")

    # Bucle principal
    while True:
        mostrar_menu()
        opcion = input("Selecciona una opción (1-4): ")

        if opcion == '1':
            consultar_saldo()
        elif opcion == '2':
            depositar()
        elif opcion == '3':
            retirar()
        elif opcion == '4':
            print("Gracias por usar el cajero. ¡Hasta luego!")
            break
        else:
            print("Opción no válida. Intenta nuevamente.")

if __name__ == "__main__":
    main()